tuf/api/metadata.py

```diff
@@ -861,22 +904,84 @@ def to_dict(self) -> Dict[str, Any]:
         }
 
 
+class TargetFile:
+    """A container with information about a particular target file.
+
+    Attributes:
+        length: An integer indicating the length of the target file.
+        hashes: A dictionary mapping hash algorithms to the
+            hashes resulting from applying them over the metadata file
+            contents::
+
+              'hashes': {
+                    '<HASH ALGO 1>': '<TARGET FILE HASH 1>',
+                    '<HASH ALGO 2>': '<TARGET FILE HASH 2>',
+                    ...
+                }
+        custom: An optional dictionary which may include version numbers,
+            dependencies, or any other data that the application wants
+            to include to describe the target file::
+                'custom': {
+                    'type': 'metadata',
+                    'file_permissions': '0644',
+                    ...
+                } // optional
+
+        unrecognized_fields: Dictionary of all unrecognized fields.
+
+    """
+
+    def __init__(
+        self,
+        length: int,
+        hashes: Dict[str, str],
+        custom: Optional[Dict[str, Any]] = None,
+        unrecognized_fields: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        self.length = length
+        self.hashes = hashes
+        self.custom = custom
+        self.unrecognized_fields: Mapping[str, Any] = unrecognized_fields or {}
+
+    def __eq__(self, other: "TargetFile") -> bool:
+        """Compare objects by their values instead of by their identities."""
+        return (
+            self.length == other.length
+            and self.hashes == other.hashes
+            and self.custom == other.custom
```

Comment:
I see `__eq__()` is still here (which is fine -- maybe you use it in tests) but it's still not complete: it should compare unrecognized_fields instead of custom